src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityCreateController.php

```diff
@@ -55,31 +53,40 @@ class EntityCreateController extends Controller
      *
      * @param Request $request
      *
-     * @param $targetEnvironment
+     * @param int $serviceId
+     * @param string $targetEnvironment
      * @return array|RedirectResponse
      */
-    public function typeAction(Request $request, $targetEnvironment)
+    public function typeAction(Request $request, $serviceId, $targetEnvironment)
     {
         $command = new ChooseEntityTypeCommand();
         $form = $this->createForm(ChooseEntityTypeType::class, $command);
 
+        $service = $this->authorizationService->getServiceById($serviceId);
+
         $form->handleRequest($request);
         if ($form->isSubmitted() && $form->isValid()) {
             // forward to create action.
-            return $this->redirectToRoute('entity_add', ['targetEnvironment' => $targetEnvironment, 'type' => $form->get('type')->getData()]);
+            return $this->redirectToRoute('entity_add', [
+                'serviceId' => $service->getId(),
+                'targetEnvironment' => $targetEnvironment,
+                'type' => $form->get('type')->getData()
+            ]);
         }
 
         return [
             'form' => $form->createView(),
+            'serviceId' => $service->getId(),
             'environment' => $targetEnvironment,
         ];
     }
 
     /**
      * @Method({"GET", "POST"})
-     * @Route("/entity/create/{type}/{targetEnvironment}", name="entity_add")
+     * @Route("/entity/create/{type}/{serviceId}/{targetEnvironment}", name="entity_add")
```

Comment:
Is it possible to keep the order: /entity/intent/{serviceId}/{other params}. This makes the routes somewhat more predictable in my opinion. Other routes that include the service id also stick to this 'convention'

Reply:
I don't think this is going to have much impact either way. So changed it.